src/main/java/uk/gov/companieshouse/confirmationstatementapi/client/OracleQueryClient.java

```diff
@@ -29,4 +29,15 @@ public Long getCompanyTradedStatus(String companyNumber) {
 
         return companyTradedStatus;
     }
+
+    public int getShareholderCount(String companyNumber) {
+        var shareholderCountUrl = String.format("%s/company/%s/shareholders/count", oracleQueryApiUrl, companyNumber);
+        LOGGER.info("Calling Oracle Query API URL (get): {}", shareholderCountUrl);
+
+        ResponseEntity<Integer> response = restTemplate.getForEntity(shareholderCountUrl, Integer.class);
+        var count = response.getBody();
+        LOGGER.info("Received {} from Oracle Query API URL (get): {}", count, shareholderCountUrl);
+
+        return count;
```

Comment:
you are returning count here which is of type Integer, but the return type for this method is int.

If somehow the response body was null, we would get a null pointer exception here.
I would suggest either setting the return type to Integer. Or add a null check within this method